udnssdk.go

```diff
@@ -267,25 +297,32 @@ func CheckAuthResponse(r *http.Response, body []byte) error {
 
 	//var er ErrorResponseList
 	var er ErrorResponse
-	//fmt.Printf("Body: %s\n", body)
+	//log.Printf("Body: %s\n", body)
 
 	err := json.Unmarshal(body, &er)
 	//err = json.NewDecoder(r.Body).Decode(errorResponse)
+
+	if err == nil {
+		er.Response = r
+		//log.Printf("CheckAuthResponse: %d", er)
+
+		return er
+	}
+	log.Printf("ERROR: %+v - Body: %s\n", err, body)
+	var er2 []ErrorResponse
+	err = json.Unmarshal(body, &er2)
+	//err = json.NewDecoder(r.Body).Decode(errorResponse)
 	if err != nil {
-		fmt.Printf("ERROR: %+v - Body: %s", err, body)
 		return err
 	}
-	er.Response = r
-	//log.Printf("CheckAuthResponse: %d", er)
-
-	return er
-
+	//log.Printf("CheckResponse: %+v", er)
+	x := &ErrorResponseList{Response: r, Responses: er2}
```

Comment:
can we just inline this in the return? or does it not like you doing the pointer there?